### PR TITLE
Improve docker-compose environment

### PR DESCRIPTION
**Pull request by mssola**
## Description

Improve the container setup for development purposes.

## Change Type

*Please select the correct option.*

- [ ] **Bug Fix** (a non-breaking change which fixes an issue)
- [x] **New Feature** (a non-breaking change which adds new functionality)
- [ ] **Documentation Update** (a change which only updates documentation)

## Checklist

*Please check off each item if the requirement is met.*

- [ ] I have verified that my code follows RMT's coding standards with `rubocop`.
- [ ] I have reviewed my own code and believe that it's ready for an external review.
- [ ] I have provided comments for any hard-to-understand code.
- [ ] I have documented the `MANUAL.md` file with any changes to the user experience.
- [ ] RMT's test coverage remains at 100%.
- [ ] If my changes are non-trivial, I have added a changelog entry to notify users at `package/obs/rmt-server.changes`.

## To do

- [ ] Make sure that the client can be called without any issues.
- [ ] Mirroring directories are volumes that can be mounted/dismounted.
- [x] Upgrade image to `opensuse/leap:15.3` and cleanup stuff.
- [ ] Make sure that these changes do not affect neither the `test` nor the `production` environments.

### CODE REVIEW

**Comment by mssola** [comment]
[Project Maintenance: improve container setup in RMT](https://trello.com/c/IqYt8oak/2071-project-maintenance-improve-container-setup-in-rmt)

**Comment by felixsch** [comment]
Hey @mssola 

Do you have a plan how to move forward with this changes? Is there anything left to do?

File: README.md [hunk 1]
```diff
@@ -66,16 +66,43 @@ Please view our [guide](docs/installation.md) to assist you in the RMT installat
...
+```
+
+Another option is to simply attach a new session into the running `rmt` service,
+which already has the needed devices and capabilities. Thus, you could do
+something like this:
+
+``` sh
+$ docker-compose exec rmt /bin/bash
+> SUSEConnect -r <regcode> http://localhost:${EXTERNAL_PORT}
+```
```
[the remark this answers is not in the document]
Comment (mssola) [reply README.md:113]: Resolved.

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/220587/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged e460a99e0ff81db89ab6c46860c3c471497fb670 into 36cce88c8cbddcdbcde9051b9dea98095183be53)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/220588/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 16aeb41f5259f6a15511972ed8f96185a63f2448 into 36cce88c8cbddcdbcde9051b9dea98095183be53)

File: config/rmt.yml [hunk 2]
```diff
@@ -3,27 +3,26 @@
...
 database_test:
   <<: *database
   database: rmt_test
 
 scc:
   host: https://scc.suse.com/connect
-  username:
-  password:
+  username: <% ENV['SCC_USERNAME'] %>
+  password: <% ENV['SCC_PASSWORD'] %>
```
Comment (lcaparroz) [inline config/rmt.yml:25]: Here, the ERB syntax is missing the equal signs:

```suggestion
  username: <%= ENV['SCC_USERNAME'] %>
  password: <%= ENV['SCC_PASSWORD'] %>
```

I took some time to figure it out, but after changing, it worked.

**Review by lcaparroz — COMMENTED** [review]
I think it's in good shape, I was able to run the whole environment following the instructions on the README.

Regarding the use of `rmt-cli` in the RMT server container, I got one problem:

The first time I ran `rmt-cli sync`, it resulted in this error:

```ruby
/srv/www/rmt/lib/suse/connect/api.rb:152:in `write': Permission denied @ rb_sysopen - /var/lib/rmt/system_uuid (Errno::EACCES)
```

The file doesn't exist when the container is built, and it also seems the default container user doesn't have the permissions.

Looking on Docker files for integration, I found a snippet on `integration/Dockerfile.test` where it creates the mentioned file at build time.

I used it to work around the problem, but it would be better if this is dealt automatically when creating the container:

```sh
dmidecode -s system-uuid > /var/lib/rmt/system_uuid
```

Other than this, the Docker setup seems to fulfill the purpose for development.

File: config/rmt.yml (hunk 2 above)
Comment (mssola) [reply config/rmt.yml]: Done!

**Comment by mssola** [comment]
> I used it to work around the problem, but it would be better if this is dealt automatically when creating the container:

Done!

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/220916/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged ee71ea2f2f99226c63e314c21fcd7641c46112cf into 36cce88c8cbddcdbcde9051b9dea98095183be53)

**Review by lcaparroz — APPROVED** [review]
Nice, it's working fine without any additional steps :+1:

**Pull request merged**